### PR TITLE
Added HTTP parser tests

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
Added first batch of parser validation tests from Kestrel codebase.
Will add second batch in a subsequent PR

### CODE REVIEW

**Comment by KrzysztofCwalina** [comment]
cc: @davidfowl

**Comment by davidfowl** [comment]
You missed these https://github.com/aspnet/KestrelHttpServer/blob/dev/test/Microsoft.AspNetCore.Server.KestrelTests/HttpParserTests.cs

**Comment by KrzysztofCwalina** [comment]
@davidfowl, thanks. i will port these tomorrow.

**Comment by KrzysztofCwalina** [comment]
@davidfowl, I added the tests to #1397

File: tests/System.Text.Http.Parser.Tests/BadHttpRequestTests.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        {
+            HttpParser parser = new HttpParser();
+            var parsed = new Request();
+            var rob = new ReadOnlyBytes(Encoding.ASCII.GetBytes(request));
+            try {
+                parser.ParseRequestLine(parsed, rob, out var consumed);
+                parser.ParseHeaders(parsed, rob.Slice(consumed), out consumed);
+            }
+            catch(BadHttpRequestException e) {
+                //Assert.Equal(expectedExceptionMessage, e.Message);
```
Comment (ahsonkhan) [inline tests/System.Text.Http.Parser.Tests/BadHttpRequestTests.cs]: Why is this commented out?

**Pull request merged**